torch/autograd/forward_ad.py

```diff
@@ -129,16 +128,15 @@ def make_dual(tensor, tangent, *, level=None):
     return torch._VF._make_dual(tensor, tangent, level=level)
 
 
-_UnpackedDualTensor = namedtuple("_UnpackedDualTensor", ["primal", "tangent"])
-
-
-class UnpackedDualTensor(_UnpackedDualTensor):
+class UnpackedDualTensor(NamedTuple):
```

Comment:
@XuehaiPan did you need to subclass NamedTuple? Why did we need to change the original code?

Reply:
We can keep it as it is. I can revert this if that is preferred.

The original code was previously changed here https://github.com/pytorch/pytorch/pull/76492/files#diff-a4efad4ba18fffdfb264c21e5475997a24a743089a899f8ec1a5ff962c6738d9 3 years ago. That was a workaround to add `__doc__` to a namedtuple type created by `collections.namedtuple`. Subclassing `typing.NamedTuple` is a more modern way to create a namedtuple type.

Reply:
no this is good, thank you for explaiing